lib/engine.py

```diff
@@ -695,10 +661,62 @@ def table_name(self, name=None, dbname=None):
                 dbname = ''
         return self.opts["table_name"].format(db=dbname, table=name)
 
+    def to_csv(self):
+        # due to Cyclic imports we can not move this import to the top
+        from retriever.lib.tools import sort_csv
+        for item in list(self.script.urls.keys()):
+            table_name = self.table_name()
+            csv_file_output = os.path.normpath(table_name + '.csv')
+            csv_file = open_fw(csv_file_output)
+            csv_writer = open_csvw(csv_file)
+            self.get_cursor()
+            self.set_engine_encoding()
+            self.cursor.execute("SELECT * FROM  {};".format(table_name))
+            row = self.cursor.fetchone()
+            colnames = [u'{}'.format(tuple_i[0]) for tuple_i in self.cursor.description]
+            csv_writer.writerow(colnames)
+            while row is not None:
+                csv_writer.writerow(row)
+                row = self.cursor.fetchone()
+            csv_file.close()
+            sort_csv(csv_file_output)
+        self.disconnect()
+
     def warning(self, warning):
         new_warning = Warning('%s:%s' % (self.script.shortname, self.table.name), warning)
         self.warnings.append(new_warning)
 
+    def load_data(self, filename):
+        """Generator returning lists of values from lines in a data file
+
+        1. Works on both delimited (csv module) fixed width data (extract_fixed_width)
+        2. Identifies the delimiter if not known
+        3. Removes extra line endings
+
+        """
+        if not self.table.delimiter:
+            self.set_table_delimiter(filename)
+
+        reg = re.compile("\\r\\n|\n|\r")
+        dataset_file = open_fr(filename)
```

Comment:
Seems like settings my the delimiter and defining reg and only used in non-fixed-width data. Should the go in the else?